src/main/java/org/embulk/input/zendesk/utils/ZendeskUtils.java

```diff
@@ -0,0 +1,167 @@
+                if (isNull(data)) {
+                    pageBuilder.setNull(column);
+                }
+                else {
+                    pageBuilder.setString(column, data.asText());
+                }
+            }
+
+            @Override
+            public void timestampColumn(Column column)
+            {
+                JsonNode data = record.get(column.getName());
+                if (isNull(data)) {
+                    pageBuilder.setNull(column);
+                }
+                else {
+                    Timestamp value = getTimestampValue(data.asText());
+                    if (value == null) {
+                        pageBuilder.setNull(column);
+                    }
+                    else {
+                        pageBuilder.setTimestamp(column, value);
+                    }
+                }
+            }
+
+            @Override
+            public void booleanColumn(Column column)
+            {
+                JsonNode data = record.get(column.getName());
+                if (isNull(data)) {
+                    pageBuilder.setNull(column);
+                }
+                else {
+                    Boolean value = getBooleanValue(data);
+                    pageBuilder.setBoolean(column, value);
+                }
+            }
+
+            @Override
+            public void longColumn(Column column)
+            {
+                JsonNode data = record.get(column.getName());
+                if (isNull(data)) {
+                    pageBuilder.setNull(column);
+                }
+                else {
+                    Long value = getLongValue(data);
+                    pageBuilder.setLong(column, value);
+                }
+            }
+
+            @Override
+            public void doubleColumn(Column column)
+            {
+                JsonNode data = record.get(column.getName());
+                if (isNull(data)) {
+                    pageBuilder.setNull(column);
+                }
+                else {
+                    Double value = getDoubleValue(data);
+                    pageBuilder.setDouble(column, value);
+                }
+            }
+        });
+        pageBuilder.addRecord();
+    }
+
+    private static boolean isNull(JsonNode jsonNode)
+    {
+        return jsonNode == null || jsonNode.isNull();
+    }
+
+    /*
+     * For getting the timestamp value of the node
+     * Sometime if the parser could not parse the value then return null
+     * */
+    private static Timestamp getTimestampValue(String value)
+    {
+        Timestamp result = null;
+        try {
+            long timeStamp = ZendeskDateUtils.isoToEpochSecond(value);
+            result = Timestamp.ofEpochSecond(timeStamp);
+        }
+        catch (Exception e) {
+            logger.warn("Error when parse time stamp data " + value);
+        }
+        return result;
+    }
+
+    private static Long getLongValue(JsonNode value)
```

Comment:
I don't think extract those getter out would bring the advantages because the getter itself e.g. `data.asLong` already well described.

Apply for `boolean`, `long` and `double`.

Reply:
fixed